src/app/review-rating/review-rating.component.ts

```diff
@@ -0,0 +1,74 @@
+import { Component, Input } from '@angular/core';
+import { Router } from '@angular/router';
+import { ModalController } from '@ionic/angular';
+import { ReviewRatingService, RatingData } from './review-rating.service';
+
+@Component({
+  selector: 'app-review-rating',
+  templateUrl: './review-rating.component.html',
+  styleUrls: ['./review-rating.component.scss']
+})
+export class ReviewRatingComponent {
+  
+  // Default redirect i.e home page.
+  redirect = ['/'];
+
+  ratingData: RatingData = {
+  	assessment_review_id: null,
+  	rating : 0.5,
+  	comment: '',
+  	tags: []
+  };
+  // variable to control the button text to indicate rating 
+  submitting = false;
+
+  constructor(
+  	private reviewRatingService: ReviewRatingService,
+  	private modalController : ModalController,
+  	private router : Router) 
+  {}
+
+  // Review ID is required if this component is to be used.upon detecting incoming/changes of value, set passed reviewId into local var
+  @Input()
+  set reviewId(reviewId: number) {
+  	this.ratingData.assessment_review_id = reviewId;    
+  }
+
+  submitReviewRating() {  	
+    this.submitting = true;
+  	// round to 2 decimal place
+  	this.ratingData.rating = +(this.ratingData.rating.toFixed(2));
+  	
+  	this.reviewRatingService.submitRating(this.ratingData).subscribe(result => {
+      this.submitting = false;
+  		this.closeReviewRating();
+  	});
+  }
+
+  closeReviewRating() {
+  	this.modalController.dismiss();
+    // if this.redirect == false, don't redirect to another page
+    if (this.redirect) {
+      this.router.navigate(this.redirect);
+    }
+  }
+
+  addOrRemoveTags(tag) {
```

Comment:
We have a function `addOrRemove()` in `utils.service` to do the same thing. Please reuse it here.